runtime/hsa-runtime/core/runtime/amd_aie_aql_queue.cpp

```diff
@@ -193,8 +221,258 @@ uint64_t AieAqlQueue::AddWriteIndexAcqRel(uint64_t value) {
 }
 
 void AieAqlQueue::StoreRelaxed(hsa_signal_value_t value) {
-  atomic::Store(signal_.hardware_doorbell_ptr, uint64_t(value),
-                std::memory_order_release);
+  std::unordered_map<uint32_t, void*> vmem_handle_mappings;
+
+  auto &driver = static_cast<XdnaDriver &>(
+      core::Runtime::runtime_singleton_->AgentDriver(agent_.driver_type));
+  if (driver.GetHandleMappings(vmem_handle_mappings) != HSA_STATUS_SUCCESS) {
+    return;
+  }
+
+  int fd = 0;
+  if (driver.GetFd(fd) != HSA_STATUS_SUCCESS) {
+    return;
+  }
+
+  SubmitCmd(hw_ctx_handle_, fd, amd_queue_.hsa_queue.base_address,
+            amd_queue_.read_dispatch_id, amd_queue_.write_dispatch_id,
+            vmem_handle_mappings);
+}
+
+hsa_status_t AieAqlQueue::SyncBos(std::vector<uint32_t> &bo_args, int fd) {
```

Comment:
Since we're not mutating `bo_args` it should be passed by `const std::vector<uint32_t>&`.